framework/src/variables/MooseVariableData.C

```diff
@@ -192,6 +192,17 @@ MooseVariableData<OutputType>::MooseVariableData(const MooseVariableFE<OutputTyp
   _grad_phi_face = &_grad_phi_face_assembly_method(_assembly, _fe_type);
 }
 
+template <typename OutputType>
+unsigned int
+MooseVariableData<OutputType>::needSolutionStates() const
```

Comment:
This function can be non-template.

Reply:
It's not a function template - it's a templated class.